### PR TITLE
Update TurkishLocale

### PR DESCRIPTION
**Pull request by beucismis**
## Pull Request Checklist

Thank you for taking the time to improve Arrow! Before submitting your pull request, please check all *appropriate* boxes:

- [x] 🧪  Added **tests** for changed code.
- [x] 🛠️  All tests **pass** when run locally (run `tox` or `make test` to find out!).
- [x] 🧹  All linting checks **pass** when run locally (run `tox -e lint` or `make lint` to find out!).
- [x] 📚  Updated **documentation** for changed code.
- [x] ⏩  Code is **up-to-date** with the `master` branch.

If you have *any* questions about your code changes or any of the points above, please submit your questions along with the pull request and we will try our best to help!

## Description of Changes

Update `TurkishLocale` class. Added `and_word`, `week`, `weeks` and `meridians`.
Edit: Add `TestTurkishLocale` class in `tests/test_test_locales.py`.

### CODE REVIEW

**Comment by beucismis** [comment]
If you give me some time I can add the `TestTurkishLocale` class. What other test class can I get a template from?

**Comment by krisfremen** [comment]
> If you give me some time I can add the `TestTurkishLocale` class. What other test class can I get a template from?

Sure, take your time. 

Take a look at the Macedonian or Tagalog locale for a more comprehensive test suite.

Cheers

**Comment by beucismis** [comment]
Cheers, thank you. I added `TestTurkishLocale` class.

**Comment by beucismis** [comment]
Hi, Some localizations have problems. I am trying to understand.
For example, "week" is output instead of "1 week".

**Comment by anishnya** [comment]
> Hi, Some localizations have problems. I am trying to understand.
> For example, "week" is output instead of "1 week".

@beucismis could you provide the list of locales where you've noticed this? From experience with dehumanize, there are some locales that need work. @krisfremen @jadchaar, we might want to open this up to be a separate issue and have an open list of locales with issues.

**Comment by krisfremen** [comment]
@anishnya agreed, we should open up issues and tackle them separately

**Comment by beucismis** [comment]
> @beucismis could you provide the list of locales where you've noticed this?

```
>>> past = arrow.utcnow().shift(weeks=-1)
>>> past.humanize(locale="tr")
'hafta önce'
```
False. It should be "1 hafta önce" or "bir hafta önce". The English translation of "hafta önce" is "week ago".

File: arrow/locales.py [hunk 1]
```diff
@@ -2311,12 +2312,16 @@ class TurkishLocale(Locale):
         "hours": "{0} saat",
         "day": "bir gün",
         "days": "{0} gün",
+        "week": "hafta",
```
Comment (beucismis) [inline arrow/locales.py:2315]: Should I write "bir hafta" instead of "hafta"?

"hafta" = "week"
"bir hafta" = "a week"

Comment (anishnya) [reply arrow/locales.py:2315]: "bir hafta" would be correct.

**Comment by beucismis** [comment]
Test results successful.

**Comment by anishnya** [comment]
Hey @beucismis. There are some extremely minor linting errors that need to be fixed. It is mainly just an issue with some of the longer assert statements within the test cases. See the output on the linter checks for more specifics. Other than that, everything else looks good to pull in. Let us know if you have any further questions, and we appreciate your contribution!

**Comment by beucismis** [comment]
@anishnya output looks correct.

```
>>> past = arrow.utcnow().shift(seconds=0)
>>> print(past.humanize(locale="tr"))
'şimdi'
```

**Review by anishnya — APPROVED** [review]

**Comment by anishnya** [comment]
Thanks for the contribution @beucismis!

**Comment by beucismis** [comment]
@anishnya thanks.

**Pull request merged**